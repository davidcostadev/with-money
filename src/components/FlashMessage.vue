<template>
  <md-snackbar :md-active.sync="show_flash_message">
    <span>{{flash_message}}</span>
    <md-button class="md-primary" @click="hide" md-persistent>Close</md-button>
  </md-snackbar>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  computed: {
    ...mapGetters([
      'flash_message',
      'show_flash_message',
    ]),
  },
  methods: {
    hide() {
      this.$store.dispatch('hideFlashMessage');
    },
  },
};
</script>
