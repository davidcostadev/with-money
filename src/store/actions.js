import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
