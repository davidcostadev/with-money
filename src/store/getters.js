export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
