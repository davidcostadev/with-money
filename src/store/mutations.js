import Moment from 'moment';
import * as types from './types';
import { updateStatesMont } from '../utils/state';

export default {
  [types.ADD_USER](state, payload) {
    state.user = payload;
  },
  [types.TO_CURRENT_MONTH](state) {
    state.state_month = state.current_month;
    updateStatesMont(state);
  },
  [types.TO_PREVIUS_MONTH](state) {
    state.state_month = Moment(state.state_month).subtract(1, 'month');
    updateStatesMont(state);
  },
  [types.TO_NEXT_MONTH](state) {
    state.state_month = Moment(state.state_month).add(1, 'month');
    updateStatesMont(state);
  },
};
