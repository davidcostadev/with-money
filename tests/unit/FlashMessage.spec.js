import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import Vuex from 'vuex';
import FlashMessage from '../../src/components/FlashMessage';
import getters from '../../src/store/getters';
import actions from '../../src/store/actions';
import mutations from '../../src/store/mutations';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);

describe('FlashMessage Component', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        flash_message: '',
      },
      getters,
      actions,
      mutations,
    });
  });

  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(FlashMessage, { store, localVue });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Beravior should be', () => {
    it('hidden', () => {
      const wrapper = mount(FlashMessage, { store, localVue });

      expect(wrapper.vm.flash_message).toEqual('');
      expect(wrapper.vm.show_flash_message).toEqual(false);
    });

    it('show', async () => {
      const wrapper = shallowMount(FlashMessage, { store, localVue });

      await wrapper.vm.$store.dispatch('showFlashMessage', 'this is the message');

      expect(wrapper.vm.flash_message).toEqual('this is the message');
      expect(wrapper.vm.show_flash_message).toEqual(true);

      expect(wrapper.element).toMatchSnapshot();
    });

    it('hide the flash message', async () => {
      const wrapper = shallowMount(FlashMessage, { store, localVue });

      await wrapper.vm.$store.dispatch('showFlashMessage', 'this is the message');

      expect(wrapper.vm.flash_message).toEqual('this is the message');
      expect(wrapper.vm.show_flash_message).toEqual(true);

      wrapper.vm.hide();

      expect(wrapper.vm.flash_message).toEqual('');
      expect(wrapper.vm.show_flash_message).toEqual(false);
    });
  });
});
