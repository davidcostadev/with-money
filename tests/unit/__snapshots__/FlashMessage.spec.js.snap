// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessage Component Beravior should be show 1`] = `
<mdsnackbar-stub
  mdactive="true"
  mdduration="4000"
  mdposition="center"
>
  <span>
    this is the message
  </span>
   
  <mdbutton-stub
    class="md-primary"
    md-persistent=""
    type="button"
  >
    Close
  </mdbutton-stub>
</mdsnackbar-stub>
`;

exports[`FlashMessage Component Render default 1`] = `
<mdsnackbar-stub
  mdduration="4000"
  mdposition="center"
>
  <span>
    
  </span>
   
  <mdbutton-stub
    class="md-primary"
    md-persistent=""
    type="button"
  >
    Close
  </mdbutton-stub>
</mdsnackbar-stub>
`;
